### PR TITLE
Added r-puniform as a package

### PR DESCRIPTION
**Pull request by thewchan**
Checklist
- [ ] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [ ] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [ ] Source is from official source.
- [ ] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [ ] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [ ] Build number is 0.
- [ ] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [ ] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [ ] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by thewchan** [comment]
@conda-forge/r @conda-forge/staged-recipes  ready for review!

File: recipes/r-puniform/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+    - r-base
+    - r-adgoftest >=0.3
+    - r-rcpp >=0.12.15
+    - r-rcpparmadillo
+    - r-metafor
+
+  run:
+    - _openmp_mutex  # [linux]
+    - libopenblas  # [unix]
+    - llvm-openmp  # [osx]
```
Comment (dbast) [inline recipes/r-puniform/meta.yaml:48]:
```suggestion
    - {{ compiler('c') }}  # [not win]
    - {{ compiler('m2w64_c') }}  # [win]
    - {{ compiler('cxx') }}  # [not win]
    - {{ compiler('m2w64_cxx') }}  # [win]
    - {{ posix }}filesystem  # [win]
    - {{ posix }}make
    - {{ posix }}sed  # [win]
    - {{ posix }}coreutils  # [win]
    - {{ posix }}zip  # [win]
    - llvm-openmp  # [osx]
    - libgomp      # [linux]

  host:
    - libblas  # [unix]
    - r-base
    - r-adgoftest >=0.3
    - r-rcpp >=0.12.15
    - r-rcpparmadillo
    - r-metafor

  run:
```

see:
* https://conda-forge.org/docs/maintainer/knowledge_base.html#openmp
* https://conda-forge.org/docs/maintainer/knowledge_base.html#blas

Comment (thewchan) [reply recipes/r-puniform/meta.yaml]: ✔️

**Comment by thewchan** [comment]
@dbast it seems that this change did not work

**Comment by thewchan** [comment]
https://github.com/orgs/conda-forge/teams/r https://github.com/orgs/conda-forge/teams/staged-recipes @dbast This is ready to be reviewed again

**Comment by dbast** [comment]
where is the blis dependency coming from? 

`INFO (r-puniform,lib/R/library/puniform/libs/puniform.so): Needed DSO lib/libblis.so.4 found in conda-forge::blis-0.9.0-h166bdaf_0`

are we maybe missing libblas in the host dependencies, which provides this via run_exports?

File: recipes/r-puniform/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+    - {{ posix }}filesystem  # [win]
+    - {{ posix }}make
+    - {{ posix }}sed  # [win]
+    - {{ posix }}coreutils  # [win]
+    - {{ posix }}zip  # [win]
+    - llvm-openmp  # [osx]
+    - libgomp      # [linux]
+
+  host:
+    - liblapack  # [unix]
```
Comment (dbast) [inline recipes/r-puniform/meta.yaml:39]:
```suggestion
    - libblas  # [unix]
    - liblapack  # [unix]
```

Comment (dbast) [reply recipes/r-puniform/meta.yaml:39]: lets try that.

File: recipes/r-puniform/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+    - libblas  # [unix]
+    - liblapack  # [unix]
+    - r-base
+    - r-adgoftest >=0.3
+    - r-rcpp >=0.12.15
+    - r-rcpparmadillo
+    - r-metafor
+
+  run:
+    - blis  # [unix]
```
Comment (dbast) [inline recipes/r-puniform/meta.yaml:48]:
```suggestion
```

Comment (thewchan) [reply recipes/r-puniform/meta.yaml]: @dbast thanks for helping to figure this out!

**Comment by thewchan** [comment]
@dbast if you're good with this can it be merged?

**Comment by dbast** [comment]
@thewchan I am good now.

**Pull request merged**